pwndbg/commands/kbase.py

```diff
@@ -48,6 +53,10 @@ def kbase() -> None:
                 )
             )
             continue
-        if b == magic:
-            print(M.success(f"Found virtual base address: {mapping.vaddr:#x}"))
+        if b in magic:
+            base = mapping.vaddr
+            if arch_name == "aarch64" and b != 0x4D:
+                # "MZ" header not found subtract 0x10000 from the address to get the kbase
+                base -= 0x10000
```

Comment:
So the base of the kernel image isn't in vmmap, only the .text image is? That seems bizarre, is that expected or a bug with `vmmap`?

If we expect to find MZ here, we should add an `if` statement actually confirming it.

Reply:
It depends on the kernel version: In older arm64 kernels, the kernel base is marked as executable, and the MZ header is present. In newer arm64 kernels, the base of the kernel image appears in vmmap but isn't marked as executable, and it is empty. I don’t know if this is due to weird gdb behavior or if that's how the new arm64 kernel is mapped.